x-pack/test/functional/apps/endpoint/feature_controls/endpoint_spaces.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common']);
+  const spacesService = getService('spaces');
+  const testSubjects = getService('testSubjects');
+  const appsMenu = getService('appsMenu');
+
+  describe('spaces', () => {
+    describe('space with no features disabled', () => {
+      before(async () => {
+        await spacesService.create({
+          id: 'custom_space',
+          name: 'custom_space',
+          disabledFeatures: [],
+        });
+      });
+
+      after(async () => {
+        await spacesService.delete('custom_space');
+      });
+
+      it('shows endpoint navlink', async () => {
+        await pageObjects.common.navigateToApp('home', {
+          basePath: '/s/custom_space',
+        });
+        const navLinks = (await appsMenu.readLinks()).map(
+          (link: Record<string, string>) => link.text
+        );
+        expect(navLinks).to.contain('EEndpoint');
+      });
+
+      it(`endpoint app shows 'Hello World'`, async () => {
+        await pageObjects.common.navigateToApp('endpoint', {
+          basePath: '/s/custom_space',
+        });
+        await testSubjects.existOrFail('welcomeTitle');
+      });
+    });
+
+    describe('space with endpoint disabled', () => {
+      before(async () => {
+        await spacesService.create({
+          id: 'custom_space',
+          name: 'custom_space',
+          disabledFeatures: ['endpoint'],
+        });
+      });
+
+      after(async () => {
+        await spacesService.delete('custom_space');
+      });
+
+      it(`doesn't show dashboard navlink`, async () => {
+        await pageObjects.common.navigateToApp('home', {
+          basePath: '/s/custom_space',
+        });
+        const navLinks = (await appsMenu.readLinks()).map(
+          (link: Record<string, string>) => link.text
```

Comment:
not that it hurts, but you probably shouldn't need to specify the type here. I removed it and the inference seems correct:

![image](https://user-images.githubusercontent.com/35559/71208081-49ed5f80-2276-11ea-9e37-ef61b018c913.png)

Makes sense because `appsMenu.readLinks` is typed like so:
![image](https://user-images.githubusercontent.com/35559/71208106-58d41200-2276-11ea-935b-40b9b8837eca.png)